src/Sulu/Bundle/MediaBundle/Infrastructure/Sulu/Content/PropertyResolver/ImageMapPropertyResolver.php

```diff
@@ -0,0 +1,155 @@
+            ? $data['hotspots']
+            : [];
+
+        $hotspots = [] !== $hotspots ? $this->resolveHotspots($hotspots, $locale, $params) : ContentView::create([], []);
+
+        $returnedParams = $params;
+        unset($returnedParams['metadata']); // TODO we may should implement a PropertyResolverAwareMetadataInterface
+
+        if (!\is_array($data)
+            || !isset($data['imageId'])
+            || !\is_numeric($data['imageId'])
+        ) {
+            return ContentView::create([
+                'image' => null,
+                'hotspots' => $hotspots->getContent(),
+            ], [
+                'imageId' => null,
+                'hotspots' => $hotspots->getView(),
+                ...$returnedParams,
+            ]);
+        }
+
+        /** @var string $resourceLoaderKey */
+        $resourceLoaderKey = $params['resourceLoader'] ?? MediaResourceLoader::getKey();
+        $imageId = (int) $data['imageId'];
+
+        return ContentView::create(
+            [
+                'image' => new ResolvableResource($imageId, $resourceLoaderKey),
+                'hotspots' => $hotspots->getContent(),
+            ],
+            [
+                'imageId' => $imageId,
+                'hotspots' => $hotspots->getView(),
+                ...$returnedParams,
+            ],
+        );
+    }
+
+    /**
+     * @param non-empty-array<array<mixed>> $hotspots
+     * @param array<string, mixed> $params
+     */
+    private function resolveHotspots(array $hotspots, string $locale, array $params): ContentView
+    {
+        $metadata = $params['metadata'] ?? null;
+        \assert($metadata instanceof FieldMetadata, 'Metadata must be set to resolve hotspots.');
+        $metadataTypes = $metadata->getTypes();
+        $content = [];
+        $view = [];
+        foreach ($hotspots as $key => $block) {
+            if (!\is_array($block) || !isset($block['type']) || !\is_string($block['type'])) {
+                continue;
+            }
+            if (!isset($block['hotspot']) || !\is_array($block['hotspot'])) {
+                continue;
+            }
+
+            $type = $block['type'];
+            $formMetadata = $metadataTypes[$type] ?? null;
+
+            if (!$formMetadata instanceof FormMetadata) {
+                $errorMessage = \sprintf(
+                    'Metadata type "%s" in "%s" not found, founded types are: "%s"',
+                    $type,
+                    $metadata->getName(),
+                    \implode('", "', \array_keys($metadataTypes)),
+                );
+
+                $this->logger->error($errorMessage);
+
+                if ($this->debug) {
+                    throw new \UnexpectedValueException($errorMessage);
+                }
+
+                $type = $metadata->getDefaultType();
+                $formMetadata = $metadataTypes[$type] ?? null;
+                if (!$formMetadata instanceof FormMetadata) {
+                    continue;
+                }
```

Comment:
as discussed with @sulu/core-team we will try to fallback to default block on none debug environments.